src/components/screens/sprint-view-screen.tsx

```diff
@@ -29,30 +27,29 @@ const SprintViewScreen = () => {
   const [allocations, setAllocations] = useState<Allocations[]>([]);
   const [projects, setProjects] = useState<Projects[]>([]);
   const [timeEntries, setTimeEntries] = useState<number[]>([]);
-  const [loading, setLoading] = useState<boolean>(true);
-  const [myTasks, setMyTasks] = useState<boolean>(true);
-  const [filter, setFilter] = useState<string>("");
-  const sprintStartDate = getSprintStart((new Date()).toISOString());
+  const [loading, setLoading] = useState(false);
+  const [myTasks, setMyTasks] = useState(true);
+  const [filter, setFilter] = useState("");
+  const todaysDate = (new Date()).toISOString()
+  const sprintStartDate = getSprintStart(todaysDate);
   const sprintEndDate = getSprintEnd((new Date()).toISOString());
+  const columns = sprintViewProjectsColumns({allocations, timeEntries, projects});
+  const setError = useSetAtom(errorAtom);
 
   /**
-   * Get project data if user is logged in otherwise endless loading
+   * Get project data if user is logged in
    */
   useEffect(() => {
     setLoading(true);
-    const fetchViewData = async () => {
-      await fetchData();
-      setLoading(false);
-    };
     if (loggedInPerson) {
-      fetchViewData();
+      fetchPersonEngagement();
     }
   }, [loggedInPerson]);
 
   /**
    * Fetch allocations, project names and time entries
    */
-  const fetchData = async () => {
+  const fetchPersonEngagement = async () => {
```

Comment:
Engagement is a bit vague and doesn't directly relate to the data being fetched.
You could call it something like fetchPersonsProjectsAndAllocationsAndTimeEntries however that's a bit of a mouthful- which is itself an indicator that perhaps this function is doing to much. 
How about splitting it into 3 different functions for fetching projects, allocations and timeEntries. Then you could have a "parent" function which called the three functions in turn (ensuring allocations and projects are available before calling time entries) and handled the loading state (rather than have the loading set for all 3 functions).

Although the current function works I think making the change I suggest improves readability and will make debugging easier in the future.

Reply:
I decided to combine these three functions into one because in opposite way we were increasing time-entries' requests to backend. What's about changing function name on "fetchPersonData" or "fetchProjectDetails"? Function itself looks simplified enough.